astropy/io/votable/tree.py

```diff
@@ -2569,18 +2587,32 @@ def is_empty(self):
         """
         return self._empty
 
-    def create_arrays(self, nrows=0, config=None):
+    def create_arrays(
+        self,
+        nrows=0,
+        config=None,
+        *,
+        colnumbers: list[int] | None = None,
+    ):
         """
         Create a new array to hold the data based on the current set
         of fields, and store them in the *array* and member variable.
         Any data in the existing array will be lost.
 
         *nrows*, if provided, is the number of rows to allocate.
+        *colnumbers*, if provided, is the list of on-disk columns to select.
+        By default, all columns are selected.
         """
         if nrows is None:
             nrows = 0
+        if colnumbers is None:
+            colnumbers = list(range(len(self.all_fields)))
```

Comment:
Why is this brand new code using `list(range(...))`?

Reply:
type consistency ! if I used a range I'd need to either make the annotation more complicated or change the code again when a type checker is enabled.

Reply:
Because the purpose of this pull request is to fix a bug, not to refactor code, then I think type consistency is a good enough argument.